src/types/uint-32.ts

```diff
@@ -1,12 +1,44 @@
-import { makeClass } from "../utils/make-class";
 import { UInt } from "./uint";
+import { BinaryParser } from "../serdes/binary-parser";
 
-const UInt32 = makeClass(
-  {
-    inherits: UInt,
-    statics: { width: 4 },
-  },
-  undefined
-);
+/**
+ * Derived UInt class for serializing/deserializing 32 bit UInt
+ */
+class UInt32 extends UInt {
+  protected static readonly width: number = 32 / 8 //4
+  static readonly defaultUInt32: UInt32 = new UInt32(Buffer.alloc(UInt32.width))
+
+  constructor(bytes: Buffer) {
+    super(bytes ?? UInt32.defaultUInt32.bytes)
+  }
+
+  static fromParser(parser: BinaryParser): UInt {
+    return new UInt32(parser.read(UInt32.width));
+  }
+
+  /**
+   * Construct a UInt32 object from a number
+   * 
+   * @param val UInt32 object or number
+   */
+  static from(val: UInt32 | number): UInt32 {
```

Comment:
Similar to other comments, I would type these out as `value` and `buffer` for clarity / readability. 

Some thoughts: https://softwareengineering.stackexchange.com/questions/72593/how-to-abbreviate-variable-names#:~:text=The%20standard%20I%20use%20is,lessens%20their%20ability%20to%20communicate.